models/notification.go

```diff
@@ -311,3 +311,14 @@ func getNotificationByID(notificationID int64) (*Notification, error) {
 
 	return notification, nil
 }
+
+// UpdateNotificationStatuses updates the statuses of all of a user's notifications that are of the currentStatus type to the desiredStatus
+func UpdateNotificationStatuses(user *User, currentStatus NotificationStatus, desiredStatus NotificationStatus) error {
+	n := &Notification{Status: desiredStatus, UpdatedBy: user.ID}
+	n.BeforeUpdate()
```

Comment:
BeforeUpdate will be invoked by xorm automatically, it's unnecessary to call it manually.